src/hooks/useExtensionsData.ts

```diff
@@ -0,0 +1,117 @@
+import { getExtensionHash } from '@colony/colony-js';
+import { useMemo } from 'react';
+
+import { supportedExtensionsConfig } from '~constants';
+import {
+  useGetColonyExtensionsQuery,
+  useGetCurrentExtensionsVersionsQuery,
+} from '~gql';
+import { InstallableExtensionData, InstalledExtensionData } from '~types';
+import { notNull } from '~utils/arrays';
+
+import useColonyContext from './useColonyContext';
+
+export enum ExtensionParamType {
+  Input = 'Input',
+  Radio = 'Radio',
+  Textarea = 'Textarea',
+}
+
+interface UseExtensionsDataReturn {
+  installedExtensionsData: InstalledExtensionData[];
+  availableExtensionsData: InstallableExtensionData[];
+  loading: boolean;
+}
+
+/**
+ * Hook for fetching extensions data available to a colony
+ * and mapping it into Installed or InstallableExtensionData object
+ */
+const useExtensionsData = (): UseExtensionsDataReturn => {
+  const { colony } = useColonyContext();
+
+  const { data, loading: extensionsLoading } = useGetColonyExtensionsQuery({
+    variables: {
+      colonyAddress: colony?.colonyAddress ?? '',
+    },
+    skip: !colony,
+    fetchPolicy: 'cache-and-network',
+  });
+  const colonyExtensions = data?.getColony?.extensions?.items?.filter(notNull);
+
+  const { data: versionsData, loading: versionsLoading } =
+    useGetCurrentExtensionsVersionsQuery({
+      fetchPolicy: 'cache-and-network',
+    });
+  const extensionVersions =
+    versionsData?.listCurrentVersions?.items?.filter(notNull);
+
+  const installedExtensionsData = useMemo<InstalledExtensionData[]>(() => {
+    if (!colonyExtensions) {
+      return [];
+    }
+
+    return colonyExtensions
+      .map<InstalledExtensionData | null>((extension) => {
+        const extensionConfig = supportedExtensionsConfig.find(
+          (e) => getExtensionHash(e.extensionId) === extension?.hash,
+        );
+
+        // Unsupported extension
+        if (!extensionConfig) {
+          return null;
+        }
+
+        const { version } =
+          extensionVersions?.find((e) => e?.extensionHash === extension.hash) ||
+          {};
+
+        return {
+          ...extensionConfig,
+          ...extension,
+          availableVersion: version ?? 1,
+        };
+      })
+      .filter(notNull);
+  }, [colonyExtensions, extensionVersions]);
+
+  const availableExtensionsData = useMemo<InstallableExtensionData[]>(() => {
```

Comment:
Do we need to memoise these? This feels to me like it belongs in a named function instead.